packages/javascript/bh-shared-ui/src/components/ExploreTable/useExploreTableRowsAndColumns.tsx

```diff
@@ -0,0 +1,195 @@
+import ExploreTableHeaderCell from './ExploreTableHeaderCell';
+
+const columnHelper = createColumnHelper<MungedTableRowWithId>();
+type DataTableProps = React.ComponentProps<typeof DataTable>;
+
+type UseExploreTableRowsAndColumnsProps = Pick<
+    ExploreTableProps,
+    'onKebabMenuClick' | 'allColumnKeys' | 'selectedColumns' | 'data'
+> & { searchInput: string };
+
+const useExploreTableRowsAndColumns = ({
+    onKebabMenuClick,
+    searchInput,
+    allColumnKeys,
+    selectedColumns,
+    data,
+}: UseExploreTableRowsAndColumnsProps) => {
+    const [sortBy, setSortBy] = useState<keyof MungedTableRowWithId>();
+    const [sortOrder, setSortOrder] = useState<'asc' | 'desc'>();
+
+    const handleSort = useCallback(
+        (sortByColumn: keyof MungedTableRowWithId) => {
+            if (sortByColumn) {
+                if (sortBy === sortByColumn) {
+                    switch (sortOrder) {
+                        case 'desc':
+                            setSortOrder('asc');
+                            break;
+                        case 'asc':
+                            setSortOrder('desc');
+                            break;
+                        default:
+                            setSortOrder('desc');
+                            break;
+                    }
+                } else {
+                    setSortBy(sortByColumn);
+                    setSortOrder('desc');
+                }
+            }
+        },
+        [sortBy, sortOrder]
+    );
+
+    const handleKebabMenuClick = useCallback(
+        (e: React.MouseEvent, id: string) => {
+            if (onKebabMenuClick) onKebabMenuClick({ x: e.clientX, y: e.clientY, id });
+        },
+        [onKebabMenuClick]
+    );
+
+    const makeColumnDef = useCallback(
+        (key: keyof MungedTableRowWithId) =>
+            columnHelper.accessor(String(key), {
+                header: () => (
+                    <ExploreTableHeaderCell
+                        sortBy={sortBy}
+                        sortOrder={sortOrder}
+                        onClick={() => handleSort(key)}
+                        headerKey={key}
+                    />
+                ),
+                cell: (info) => {
+                    const value = info.getValue() as EntityField['value'];
+
+                    if (typeof value === 'boolean') {
+                        return (
+                            <div className='h-full w-full flex justify-center items-center text-center'>
+                                <FontAwesomeIcon
+                                    icon={value ? faCheck : faCancel}
+                                    color={value ? 'green' : 'lightgray'}
+                                    className='scale-125'
+                                />
+                            </div>
+                        );
+                    }
+
+                    return format({ keyprop: String(key), value, label: String(key) }) || '--';
```

Comment:
I typically see `.toString()` used so I took a look at the difference between that and `String()` and it looks like one big difference is that `String()` will gracefully handle undefined and null values (as 'null' and 'undefined'), is that by chance a case youre expecting?

Reply:
Perhaps pulling '--' into a const would make it easier to test against? And then if we need to change that fallback into a different `ReactNode` we could change it in one place!